app/src/main/java/com/hedvig/app/feature/embark/passages/audiorecorder/AudioRecorderFragment.kt

```diff
@@ -55,16 +57,35 @@ class AudioRecorderFragment : Fragment() {
                     viewState = state,
                     startRecording = ::askForPermission,
                     clock = clock,
-                    stopRecording = model::stopRecording,
-                    submit = { submitAudioRecording(state, parameters) },
-                    redo = model::redo,
-                    play = model::play,
-                    pause = model::pause,
+                    stopRecording = {
+                        model::stopRecording
+                        logWithStoryAndStore(hAnalytics::embarkAudioRecordingStopped)
+                    },
+                    submit = {
+                        submitAudioRecording(state, parameters)
+                        logWithStoryAndStore(hAnalytics::embarkAudioRecordingSubmitted)
+                    },
+                    redo = {
+                        model.redo()
+                        logWithStoryAndStore(hAnalytics::embarkAudioRecordingRetry)
+                    },
+                    play = {
+                        model.play()
+                        logWithStoryAndStore(hAnalytics::embarkAudioRecordingPlayback)
+                    },
+                    pause = {
+                        model.pause()
+                        logWithStoryAndStore(hAnalytics::embarkAudioRecordingStopped)
+                    },
```

Comment:
By putting them here, there's only the slight "problem" that if any of these are called at a state that is "illegal" the event would get triggered but the viewmodel function may reject that action, like doing "play" but the audio player isn't ready yet.
Probably the vast majority of the times the event will be correct though, so probably this is fine.